Functional.Common.Extensions/EnumerableExtensions.cs

```diff
@@ -8,6 +8,12 @@ namespace Functional
 {
     public static class EnumerableExtensions
     {
+	    public static void Apply<T>(this IEnumerable<T> source, Action<T> action)
```

Comment:
As much as I don't like the Apply function (since it is explicitly for the purpose of side effecting), we do have it for Result, Option, and Union, so it probably makes sense. If we add this, should we also add Do()  though?

Reply:
I'm the wrong person to ask, as I avoid `Do` like the plague :P

I just find `Apply` cleaner than the following:
```
foreach (var item in someLongChainOfProcessingForSomeCollection)
{
    // do stuff
}
```

I understand that's a minor difference haha.